module/abci.go

```diff
@@ -10,38 +10,116 @@ import (
 	"github.com/strangelove-ventures/poa"
 )
 
-// BeginBlocker updates the validator set without applying updates.
-// Since this module depends on staking, that module will `ApplyAndReturnValidatorSetUpdates` from x/staking.
+// We no longer have instant unbonding to support adding new validators into the set.
+// This could be tweaked to properly handle jailing / instant unbonding, but not a requirement.
+// They stop contributing to BFT consensus anyways.
 func (am AppModule) BeginBlocker(ctx context.Context) error {
 	sdkCtx := sdk.UnwrapSDKContext(ctx)
 	defer telemetry.ModuleMeasureSince(poa.ModuleName, sdkCtx.BlockTime(), telemetry.MetricKeyBeginBlocker)
 
+	if sdkCtx.BlockHeight() <= 1 {
+		return nil
+	}
+
+	valUpdates, err := am.keeper.GetStakingKeeper().GetValidatorUpdates(ctx)
+	if err != nil {
+		return err
+	}
+
+	if len(valUpdates) != 0 {
+		vals, err := am.keeper.GetStakingKeeper().GetAllValidators(ctx)
+		if err != nil {
+			return err
+		}
+
+		for _, val := range vals {
+			valBz, err := sdk.ValAddressFromBech32(val.OperatorAddress)
+			if err != nil {
+				return err
+			}
+
+			if val.Status == stakingtypes.Bonded {
```

Comment:
TODO: add note this is to make ABCI events happy so we always control the key (which changes over blocks)